nsfc-plugin/src/main/java/org/osc/controller/nsfc/utils/RedirectionApiUtils.java

```diff
@@ -184,84 +284,18 @@ public void validatePPGList(List<NetworkElement> portPairGroups) {
         }
     }
 
-    public void validateAndClear(ServiceFunctionChainEntity sfc) {
-     // TODO (Dmitry) Implement
-    }
-
-    /**
-     * TODO placeholder method while transitioning to non-db implementation
-     *
-     * Throw exception message in the format "null passed for 'type'!"
-     */
-    public void throwExceptionIfNull(Object object, Class<?> clazz) {
-        throwExceptionIfNull(object, clazz.getName());
+    public FlowClassifier buildFlowClassifier(String inspectedPortIp, ServiceFunctionChainEntity sfcEntity) {
+        FlowClassifier flowClassifier;
+        String sourcePortId = sfcEntity.getPortPairGroups().get(0).getPortPairs().get(0).getIngressPort().getElementId();
```

Comment:
sourceport and destinationport id should be inspectedPortId(not ip).

you dont need the ingress and egress ports of the PPG.

I think this will also elimnate the need for fetchSFCWithAllDepends